pkg/provision/source.go

```diff
@@ -30,6 +30,13 @@ type ReservationPoller interface {
 	Poll(nodeID pkg.Identifier, from uint64) (reservations []*Reservation, lastID uint64, err error)
 }
 
+// ReservationFromSource wraps a reservation type and has
+// a boolean to indicate it is the last reservation
+type ReservationFromSource struct {
```

Comment:
I would rename this type to something a bit more explicit:
```suggestion
type ProvisionJob struct {
```